azure/src/wizard/StorageAccountCreateStep.ts

```diff
@@ -47,6 +50,57 @@ export class StorageAccountCreateStep<T extends types.IStorageAccountWizardConte
         ext.outputChannel.appendLog(createdStorageAccount);
     }
 
+    protected async generateRelatedName(wizardContext: T, name: string, namingRules: IAzureNamingRules | IAzureNamingRules[]): Promise<string | undefined> {
+        if (!isArray(namingRules)) {
+            namingRules = [namingRules];
+        }
+
+        let preferredName: string = namingRules.some((n: IAzureNamingRules) => !!n.lowercaseOnly) ? name.toLowerCase() : name;
+
+        for (let invalidCharsRegExp of namingRules.map((n: IAzureNamingRules) => n.invalidCharsRegExp)) {
+            // Ensure the regExp uses the 'g' flag to replace _all_ invalid characters
+            invalidCharsRegExp = new RegExp(invalidCharsRegExp, 'g');
+            preferredName = preferredName.replace(invalidCharsRegExp, '');
+        }
+
+        const minLength: number = Math.max(...namingRules.map((n: IAzureNamingRules) => n.minLength));
+        const maxLength: number = Math.min(...namingRules.map((n: IAzureNamingRules) => n.maxLength));
+
+        const maxTries: number = 5;
+        let count: number = 0;
+        let newName: string;
+        while (count < maxTries) {
+            newName = this.generateSuffixedName(preferredName, minLength, maxLength);
+            if (await this.isRelatedNameAvailable(wizardContext, newName)) {
+                return newName;
+            }
+            count += 1;
+        }
+
+        return undefined;
+    }
+
+    protected async isRelatedNameAvailable(wizardContext: T, name: string): Promise<boolean> {
+        return await ResourceGroupListStep.isNameAvailable(wizardContext, name);
+    }
+
+    private generateSuffixedName(preferredName: string, minLength: number, maxLength: number): string {
+        const suffix: string = (Math.floor(Math.random() * (100 - 1)) + 1).toString();
```

Comment:
I don't see it in our azure package, but our other packages use a `getRandomHexString` to generate a suffix. I think it's more guaranteed to not have conflicts rather than a random number between 1-100.

```
export function getRandomHexString(length: number = 6): string {
        const buffer: Buffer = crypto.randomBytes(Math.ceil(length / 2));
        return buffer.toString('hex').slice(0, length);
    }
```

Reply:
Thanks Nathan! Gonna implement that one as the other packages